crates/types/src/invocation/mod.rs

```diff
@@ -1307,6 +1331,59 @@ mod serde_hacks {
     }
 }
 
+mod dto {
+    use super::{VirtualObjectHandlerType, WorkflowHandlerType};
+
+    #[derive(PartialEq, Eq, bilrost::Enumeration)]
+    enum InvocationTargetTypeKind {
+        Service = 0,
+        VirtualObject = 1,
+        Workflow = 2,
+    }
+
+    #[derive(bilrost::Message)]
+    pub(super) struct InvocationTargetType {
+        kind: InvocationTargetTypeKind,
+        shared: bool,
+    }
```

Comment:
Being future proof here, i think it might make more sense to have just `InvocationTargetTypeKind` with the 5 variants flattened than this `shared` bool.

Reply:
Sounds like a good idea. Will do that.